core/src/main/java/com/scalar/db/io/Column.java

```diff
@@ -0,0 +1,177 @@
+package com.scalar.db.io;
+
+import java.nio.ByteBuffer;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+/**
+ * An abstraction for a column.
+ *
+ * <p>This class and its implementation classes of it are primarily for internal use. Breaking
+ * changes can and will be introduced to them. Users should not depend on them.
+ */
+public interface Column<T> extends Comparable<Column<T>> {
```

Comment:
I added `Column<T>` and its implementation classes `BooleanColumn`, `IntColumn`, `BigIntColumn`, `FloatColumn`, `DoubleColumn`, `TextColumn`, and `BlobColumn`.